### DISCUSSION TITLE
What happened to the smart reply feature?

### DISCUSSION THREAD

**Discussion by eladgel** [invertase/react-native-firebase#4629] [General]
I was about to add it to our app, and it disappeared...
https://www.npmjs.com/package/@react-native-firebase/ml-natural-language

**Comment by mikehardy** [comment]
With apologies, going to copy paste:

https://github.com/invertase/react-native-firebase/issues/4612#issuecomment-733724151

> Apologies - the packages will be added later. It is on me to carry the existing "on device inference" implementations from react-native-firebase over there and get them published, but I've been called on for a couple other tasks so I have not had the time yet. Until then, if you require on-device inference you are unfortunately not able to update to the v10 releases yet (actually, the version set just before them is off limits even). I still have hopes of getting those packages up and live within the next week or two until then I can only plead for patience.

I was tapped to do some work on FlutterFire in place of focus on react-native-mlkit but I'm nearly done with that and I can come back to it. In the meantime - use the older versions here and in general the same APIs will carry over to react-native-mlkit, first release there is not intended to be a big breaking change, just moving the code to a new home / package

**Reply by eladgel** [reply]
Ping 😀

**Reply by mikehardy** [reply]
Still in the plan but still nothing to show - sorry - all my FlutterFire PRs were merged in January though which at least popped those off the stack, then I was able to roll through a stack of issues here and kick out releases last week. There is one bit of entropy surrounding firebase.database()...once() usage and the database emulator #4854 that I'm working through but I'm getting closer to having time for MLKit. Sorry for the delay, the ML-using crowd here (including yourself) has been very patient.

**Comment by eladgel** [comment]
I can start a pr.. 
It will take me a month with the load I have write now, but maybe we will make more progress...

**Reply by mikehardy** [reply]
That's a thought, but start a PR to do what? Add it back here? That's a non-starter. The on-device ML APIs are *no longer in firebase*. That means they should no longer be in react-native-firebase, which is why they aren't.

The "PR" would then be to stand up a new monorepo for the new home of Google's on-device ML APIs ("Google MLKit") and "just move the code over". Note that we already have react-native-mlkit npmjs.com namespace reserved for this as well as github space, but the github space is still a private repo and we haven't published it yet.

So...not really a PR :-). More like a new project. Just getting started with it to do the monorepo infra is the real task, the code from before should work fine once it has a home.

**Reply by eladgel** [reply]
I understand, willing to help and pr to the new project once it's up and running, just to get it going :)
